src/data-management/fake-data/fake-data-service.ts

```diff
@@ -0,0 +1,89 @@
+import { Note, StorageService } from '../types';
+
+export class FakeDataService implements StorageService {
+    private notes: Note[];
+
+    constructor() {
+        this.notes = [];
+        const date = new Date();
+        for (let i = 0; i < 3; i++) {
+            const note: Note = {
+                id: (Date.now() + i).toString(),
+                title: `Note ${i}`,
+                description: 'Lorem Ipsum Dolor Sit Amet',
+                createdAt: date,
+                modifiedAt: date,
+                isPinned: false,
+            };
+            this.notes.push(note);
+        }
+    }
+
+    async getAllNotes(): Promise<Note[]> {
+        return new Promise((resolve) => {
+            resolve(this.notes);
+        });
+    }
+
+    async getNoteById(id: string): Promise<Note | undefined> {
+        return new Promise((resolve, reject) => {
+            const note = this.notes.find((note) => note.id === id);
+            if (note) {
+                resolve(note);
+            } else {
+                reject(`No such a note in the storage. Id: ${id}`);
+            }
+        });
+    }
+
+    async addNote(newNote: Pick<Note, 'title' | 'description'>): Promise<Note> {
+        return new Promise((resolve) => {
+            const createdAt = new Date();
+            const note: Note = {
+                id: Date.now().toString(),
+                createdAt,
+                modifiedAt: createdAt,
+                isPinned: false,
+                ...newNote,
+            };
+            this.notes.push(note);
```

Comment:
Considering react expects things to be immutable such mutations will cause a problem, because it between rerenders it is going to be the same array.
It's better to do
this.notes = [...this.notes, newNote];

The same goes to all methods that mutate `this.notes`